### PR TITLE
Implement detection of keyboard navigation

### PR DESCRIPTION
**Pull request by piofinn**
## 📥 Proposed changes

Implement detection of whether the user is navigating with the mouse or keyboard, and signify this with a data-attribute on the html element.

## ☑️ Submission checklist

-   [x] I have read the [CONTRIBUTING]() doc
-   [x] `yarn build` works locally with my changes
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] I have added necessary documentation (if appropriate)

## 💬 Further comments

I've placed this in utils, but it could presumably also be in core.

### CODE REVIEW

File: packages/utils/src/tabListener.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const navKeys = ["Tab", "ArrowUp", "ArrowRight", "ArrowDown", "ArrowLeft"];
```
Comment (Saegrov) [inline packages/utils/src/tabListener.ts]: Do we alle the arrow-keys? I don't think you can use arrow keys to change focused item?

File: packages/utils/src/tabListener.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+let mousenavigation = false;
+
+function handleMouseDown() {
+    if (!mousenavigation) {
+        mousenavigation = true;
+        const htmlElement = document.querySelector("html");
+        htmlElement && htmlElement.setAttribute("data-mousenavigation", "true");
+        window.removeEventListener("mousedown", handleMouseDown);
+        window.addEventListener("keydown", handleKeydown);
```
Comment (Saegrov) [inline packages/utils/src/tabListener.ts]: Use document to add event listeners:
https://stackoverflow.com/a/2671217

File: packages/utils/src/tabListener.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            mousenavigation = false;
+            const htmlElement = document.querySelector("html");
+            htmlElement && htmlElement.removeAttribute("data-mousenavigation");
+            window.removeEventListener("keydown", handleKeydown);
+            window.addEventListener("mousedown", handleMouseDown);
+        }
+    }
+}
+
+export function initTabListener() {
```
Comment (Saegrov) [inline packages/utils/src/tabListener.ts]: I wrote a test that you might consider adding:
```typescript jsx
import { fireEvent } from "@testing-library/dom";
import { initTabListener } from "./tabListener";

let mouseNavigation = "data-mousenavigation";

test("should initial mouseNavigation set to null before clicks or tabs", () => {
    initTabListener();

    const htmlElement = document.querySelector("html");

    if (!htmlElement) {
        throw 'Something went wrong  - couldn\'t find body element';
    }

    expect(htmlElement.getAttribute(mouseNavigation)).toBe(null);
});

test("should add data-mousenavigation attributen when clicking a button", () => {
    initTabListener();

    const bodyElement = document.querySelector("body");
    const htmlElement = document.querySelector("html");

    if (!bodyElement || !htmlElement) {
        throw 'Something went wrong  - couldn\'t find body element';
    }

    const buttonElement = document.createElement("button");
    bodyElement.appendChild(buttonElement);

    fireEvent(buttonElement, new MouseEvent("mousedown", { bubbles: true }));

    expect(htmlElement.getAttribute(mouseNavigation)).toBe("true");
});

test("should remove data-mousenavigation attribute when tabbing after a click", () => {
    initTabListener();

    const bodyElement = document.querySelector("body");
    const htmlElement = document.querySelector("html");

    if (!bodyElement || !htmlElement) {
        throw 'Something went wrong  - couldn\'t find body element';
    }

    const buttonElement = document.createElement("button");
    bodyElement.appendChild(buttonElement);

    fireEvent(buttonElement, new MouseEvent("mousedown", { bubbles: true }));

    expect(htmlElement.getAttribute(mouseNavigation)).toBe("true");

    fireEvent(document, new KeyboardEvent("keydown", { key: "Tab", bubbles: true }));

    expect(htmlElement.getAttribute(mouseNavigation)).toBe(null);
});

```

File: packages/utils/src/index.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
 import { StoryTemplate } from "./internal/StoryTemplate";
+import { initTabListener } from "./tabListener";
```
Comment (Saegrov) [inline packages/utils/src/index.ts]: IMO this should be part of core and not util.

File: packages/utils/src/tabListener.ts (hunk 3 above)
Comment (Saegrov) [reply packages/utils/src/tabListener.ts]: Oh. And we need to install @testing-library/dom if we use this test.

File: packages/utils/src/tabListener.ts (hunk 1 above)
Comment (piofinn) [reply packages/utils/src/tabListener.ts]: They are used to navigate radio buttons, tabs, dropdowns and a few other structures: https://webaim.org/techniques/keyboard/#testing

File: packages/utils/src/index.ts (hunk 4 above)
Comment (piofinn) [reply packages/utils/src/index.ts]: Moved over now, but we should refactor the entire `core` package sometime soon.

File: packages/utils/src/tabListener.ts (hunk 3 above)
Comment (piofinn) [reply packages/utils/src/tabListener.ts]: Nice tests! They are now shamelessly stolen 😅

**Pull request merged**